taca/backup/backup.py

```diff
@@ -0,0 +1,309 @@
+"""Backup methods and utilities"""
+import couchdb
+import logging
+import os
+import re
+import sys
+import shutil
+import subprocess as sp
+
+from datetime import datetime
+from taca.utils.config import CONFIG
+from taca.utils import filesystem, misc
+
+logger = logging.getLogger(__name__)
+
+class run_vars(object):
+    """A simple variable storage class"""
+    def __init__(self, run):
+        self.abs_path = os.path.abspath(run)
+        self.path, self.name = os.path.split(self.abs_path)
+        self.name = self.name.split('.', 1)[0]
+        self.zip = "{}.tar.gz".format(self.name)
+        self.key = "{}.key".format(self.name)
+        self.key_encrypted = "{}.key.gpg".format(self.name)
+        self.zip_encrypted = "{}.tar.gz.gpg".format(self.name)
+
+class backup_utils(object):
+    """A class object with main utility methods related to backing up"""
+
+    def __init__(self, run=None):
+        self.run = run
+        self.fetch_config_info()
+
+    def fetch_config_info(self):
+        """Try to fecth required info from the config file. Log and exit if any neccesary info is missing"""
+        try:
+            self.data_dirs = CONFIG['backup']['data_dirs']
+            self.archive_dirs = CONFIG['backup']['archive_dirs']
+            self.keys_path = CONFIG['backup']['keys_path']
+            self.gpg_receiver = CONFIG['backup']['gpg_receiver']
+            self.check_demux = CONFIG.get('backup', {}).get('check_demux', False)
+            if self.check_demux:
+                self.couch_info = CONFIG['statusdb']
```

Comment:
@vezzi : Is this better ? As `statusDB` info is required when we wanted to check if a given run is demultiplexed from a `NAS`. So if some one run without `check_demux : True` in their config file, it will not expect for `statusdb` nor it will perform the demux check.